client/src/js/services/journal/journalColumnConfig.js

```diff
@@ -0,0 +1,130 @@
+angular.module('bhima.services')
+.service('JournalColumnConfigService', JournalColumnConfigService);
+
+JournalColumnConfigService.$inject = ['uiGridConstants', 'AppCache'];
+
+/**
+ * Posting Journal Column config Service 
+ * This service is responsible of Showing and Hiding column
+*/
+function JournalColumnConfigService(uiGridConstants, AppCache) { 
+  var service = this; 
+  
+  // Variable used to track and share the current grids API object
+  var gridApi;
+
+  //An appcache instance
+  //@todo use ui-grid-saveState
+  var cache = AppCache('columnConfigService');
+
+  //default visibility configuartion
+  var defaultVisibility = [ 
```

Comment:
Do we need the `displayName` property?  It seems to me this is an unnecessary property that will cause confusion in the future if we update the controller but forget to update the grid.

It seems to me that a cleaner implementation would be a simple JSON object that maps `field`s to booleans.  For example:
```js
// maps column fields to their visibility value.  `true` are visible by default, `false` are hidden.
var defaultVisibility = {
  'uuid' : false,
  'project_name' :  false,,
  // etc...
  'trans_date' : true
};
```
What do you think?